File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int ptr1 = 0;
        int ptr2 = numbers.length-1;
        int answer[]=new int [2];
        for(int i =0 ; i < numbers.length; i++){
            if(target == numbers[ptr1]+numbers[ptr2]){
                answer[0]=ptr1+1;
                answer[1]=ptr2+1;
                break;
            }
            else if(target > numbers[ptr1]+numbers[ptr2]){
                ptr1++;
            }
            else if(target < numbers[ptr1]+numbers[ptr2]){
                ptr2--;
            }    
        }

        
        return answer;

    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> targets;
        map<int, int> mp;
        signed long long diff;

        for (unsigned int i =0; i< nums.size(); i++){
            diff = target-nums[i];
            mp[diff] = i;
        }

        for (unsigned int i =0; i< nums.size(); i++){
            if (mp.find(nums[i]) != mp.end()) {
                if(i!=mp[nums[i]]){
                    targets.push_back(i);
                    targets.push_back(mp[nums[i]]);
                    break;
                }
            }
        }

        return targets;
    }
};
